monkey/infection_monkey/exploit/tools/helpers.py

```diff
@@ -20,6 +22,13 @@ def get_target_monkey_by_os(is_windows, is_32bit):
     )
 
 
+def get_random_file_suffix() -> str:
+    character_set = list(string.ascii_letters + string.digits + "_" + "-")
+    safe_random = SystemRandom()
+    random_string = "".join(safe_random.choices(character_set, k=8))
+    return random_string
```

Comment:
This can block indefinitely on Linux if there isn't sufficient entropy to feed `/dev/urandom`. Since this doesn't need to be cryptographically secure, we can just use `random.choices()`.

```suggestion
def get_random_file_suffix() -> str:
    character_set = list(string.ascii_letters + string.digits + "_" + "-")
    return "".join(random.choices(character_set, k=8))
```

Reply:
We are using `SystemRandom` in the Hadoop exploiter also.